clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -928,6 +928,78 @@ float3 exp2(float3);
 _HLSL_BUILTIN_ALIAS(__builtin_elementwise_exp2)
 float4 exp2(float4);
 
+//===----------------------------------------------------------------------===//
+// firstbithigh builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn T firstbithigh(T Val)
+/// \brief Returns the location of the first set bit starting from the highest
+/// order bit and working downward, per component.
+/// \param Val the input value.
+
+#ifdef __HLSL_ENABLE_16_BIT
+_HLSL_AVAILABILITY(shadermodel, 6.2)
```

Comment:
I don't think the int16_t, uint16_t, int64_t, and uint64_t cases are correct.

Reply:
To be clear, I believe we should *have* these cases, but the return types should always be `uint32_t` or vectors thereof